### PR TITLE
logging microservice added

### PR DESCRIPTION
**Pull request by Evgeny-Gromyko**

### CODE REVIEW

File: Services/LogService/ModsenOnlineStore.LogService.API/Program.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+var builder = WebApplication.CreateBuilder(args);
+
+
+// not working, error "Some services are not able to be constructed"
+//builder.Services.AddScoped<ILoggerProvider, DBLoggerProvider>(); 
+
+//how to pass ILoggerProvider inside DBLoggerProvider?
+
+//builder.Services.AddSingleton<DataContext>();
+//builder.Services.AddSingleton<DbContextOptions>(); not Working
```
Comment (elv1sabene) [inline Services/LogService/ModsenOnlineStore.LogService.API/Program.cs]: delete unnecessary comments please

File: Services/LogService/ModsenOnlineStore.LogService.API/Controllers/LogController.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+using Microsoft.Extensions.Logging;
+using ModsenOnlineStore.LogService.Domain.Entities;
+using System;
+using static System.Runtime.InteropServices.JavaScript.JSType;
+
+namespace ModsenOnlineStore.LogService.API.Controllers
+{
+    [Route("[controller]")]
+    [ApiController]
+    public class LogController : ControllerBase
```
Comment (elv1sabene) [inline Services/LogService/ModsenOnlineStore.LogService.API/Controllers/LogController.cs:10]: do you really need logging as API?

File: Services/LogService/ModsenOnlineStore.LogService.Infrastructure/Extensions/DBLoggerExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+using ModsenOnlineStore.LogService.Infrastructure.Services;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ModsenOnlineStore.LogService.Infrastructure.Extensions
+{
+    public static class DBLoggerExtensions
```
Comment (elv1sabene) [inline Services/LogService/ModsenOnlineStore.LogService.Infrastructure/Extensions/DBLoggerExtensions.cs]: do you need this class?

**Review by elv1sabene — COMMENTED** [review]
please refactor code from unnecessary comments and empty lines

File: Services/LogService/ModsenOnlineStore.LogService.API/Program.cs (hunk 1 above)
Comment (Evgeny-Gromyko) [reply Services/LogService/ModsenOnlineStore.LogService.API/Program.cs]: removed

File: Services/LogService/ModsenOnlineStore.LogService.Infrastructure/Extensions/DBLoggerExtensions.cs (hunk 3 above)
Comment (Evgeny-Gromyko) [reply Services/LogService/ModsenOnlineStore.LogService.Infrastructure/Extensions/DBLoggerExtensions.cs]: No longer needed, removed

File: Services/LogService/ModsenOnlineStore.LogService.API/Controllers/LogController.cs (hunk 2 above)
Comment (Evgeny-Gromyko) [reply Services/LogService/ModsenOnlineStore.LogService.API/Controllers/LogController.cs:10]: Maybe not, but it's a well-known approach

File: Services/LogService/ModsenOnlineStore.LogService.Domain/Entities/Log.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+{
+    public class Log
+    {
+        public int Id { get; set; }
+        public string logLevel { get; set; } = String.Empty;  // Trace/Debug/Information/Warning/Error/Critical/None
+        public int eventId { get; set; }                      // = Convert.ToInt32(MyLogEvents.GetItems);
+        // public Exception exception { get; set; } // not in use now
+        public DateTime dateTime { get; set; }
+        public string message { get; set; } = String.Empty;
+        public int userId { get; set; }
```
Comment (elv1sabene) [inline Services/LogService/ModsenOnlineStore.LogService.Domain/Entities/Log.cs]: why not PascalCase?

File: Services/LogService/ModsenOnlineStore.LogService.Domain/Entities/Log.cs [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+using System.Threading.Tasks;
+
+namespace ModsenOnlineStore.LogService.Domain.Entities
+{
+    public class Log
+    {
+        public int Id { get; set; }
+        public string logLevel { get; set; } = String.Empty;  // Trace/Debug/Information/Warning/Error/Critical/None
+        public int eventId { get; set; }                      // = Convert.ToInt32(MyLogEvents.GetItems);
+        // public Exception exception { get; set; } // not in use now
```
Comment (elv1sabene) [inline Services/LogService/ModsenOnlineStore.LogService.Domain/Entities/Log.cs]: delete comments

File: Services/LogService/ModsenOnlineStore.LogService.Infrastructure/Services/DBLoggerProvider.cs [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+        //categoryName is unnecessary but required by ILoggerProvider parameter
+        public ILogger CreateLogger(string categoryName) 
+        {
+            return new DBLogger(repository);
+        }
+
+        public void Dispose() { }
+    }
+
+
```
Comment (elv1sabene) [inline Services/LogService/ModsenOnlineStore.LogService.Infrastructure/Services/DBLoggerProvider.cs]: delete empty lines please

**Review by elv1sabene — APPROVED** [review]

File: Services/LogService/ModsenOnlineStore.LogService.Domain/Entities/Log.cs (hunk 4 above)
Comment (Evgeny-Gromyko) [reply Services/LogService/ModsenOnlineStore.LogService.Domain/Entities/Log.cs]: corrected

File: Services/LogService/ModsenOnlineStore.LogService.Domain/Entities/Log.cs (hunk 5 above)
Comment (Evgeny-Gromyko) [reply Services/LogService/ModsenOnlineStore.LogService.Domain/Entities/Log.cs]: deleted

File: Services/LogService/ModsenOnlineStore.LogService.Infrastructure/Services/DBLoggerProvider.cs (hunk 6 above)
Comment (Evgeny-Gromyko) [reply Services/LogService/ModsenOnlineStore.LogService.Infrastructure/Services/DBLoggerProvider.cs]: corrected

File: Services/Store/ModsenOnlineStore.Store.Application/Middleware/ExceptionHandlerMiddleware.cs [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+            catch (Exception ex)
+            {
+                var exceptionThrowingСontext = ex.StackTrace.Remove(0, "   at ".Length);
+                exceptionThrowingСontext = exceptionThrowingСontext.Substring(0, exceptionThrowingСontext.IndexOf("at") - 2);
+
+                var exceptionMessage = ex.InnerException;
+
+                Console.WriteLine(ex);
+
+                //send request lo LogService
```
Comment (elv1sabene) [inline Services/Store/ModsenOnlineStore.Store.Application/Middleware/ExceptionHandlerMiddleware.cs]: so what here?

File: Services/Store/ModsenOnlineStore.Store.Application/Middleware/ExceptionHandlerMiddleware.cs [hunk 8]
```diff
@@ -0,0 +1,38 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.EntityFrameworkCore.Metadata.Internal;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
```
Comment (elv1sabene) [inline Services/Store/ModsenOnlineStore.Store.Application/Middleware/ExceptionHandlerMiddleware.cs]: delete unnecessary usings

File: Services/LogService/ModsenOnlineStore.LogService.Infrastructure/Services/LogServise.cs [hunk 9]
```diff
@@ -0,0 +1,40 @@
+﻿using ModsenOnlineStore.Common;
+using ModsenOnlineStore.LogService.Application.Interfaces;
+using ModsenOnlineStore.LogService.Domain.Entities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
```
Comment (elv1sabene) [inline Services/LogService/ModsenOnlineStore.LogService.Infrastructure/Services/LogServise.cs]: usings

File: Services/LogService/ModsenOnlineStore.LogService.Infrastructure/Services/DBLogger.cs [hunk 10]
```diff
@@ -0,0 +1,41 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using ModsenOnlineStore.LogService.Application.Interfaces;
+using ModsenOnlineStore.LogService.Domain.Entities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
```
Comment (elv1sabene) [inline Services/LogService/ModsenOnlineStore.LogService.Infrastructure/Services/DBLogger.cs]: usings

File: Services/LogService/ModsenOnlineStore.LogService.Infrastructure/Data/LogRepository.cs [hunk 11]
```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using ModsenOnlineStore.LogService.Application.Interfaces;
+using ModsenOnlineStore.LogService.Domain.Entities;
```
Comment (elv1sabene) [inline Services/LogService/ModsenOnlineStore.LogService.Infrastructure/Data/LogRepository.cs:3]: check usings

File: Services/LogService/ModsenOnlineStore.LogService.Domain/Entities/Log.cs [hunk 12]
```diff
@@ -0,0 +1,31 @@
+﻿using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
```
Comment (elv1sabene) [inline Services/LogService/ModsenOnlineStore.LogService.Domain/Entities/Log.cs]: the same

File: Services/LogService/ModsenOnlineStore.LogService.API/Controllers/LogController.cs [hunk 13]
```diff
@@ -0,0 +1,71 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using ModsenOnlineStore.LogService.Application.Interfaces;
+using ModsenOnlineStore.LogService.Domain.Entities;
+using ModsenOnlineStore.LogService.Infrastructure.Services;
+using System;
+using static System.Runtime.InteropServices.JavaScript.JSType;
```
Comment (elv1sabene) [inline Services/LogService/ModsenOnlineStore.LogService.API/Controllers/LogController.cs]: check usings

File: Services/Store/ModsenOnlineStore.Store.Application/Middleware/ExceptionHandlerMiddleware.cs (hunk 8 above)
Comment (Evgeny-Gromyko) [reply Services/Store/ModsenOnlineStore.Store.Application/Middleware/ExceptionHandlerMiddleware.cs]: removed

File: Services/LogService/ModsenOnlineStore.LogService.Domain/Entities/Log.cs (hunk 12 above)
Comment (Evgeny-Gromyko) [reply Services/LogService/ModsenOnlineStore.LogService.Domain/Entities/Log.cs]: removed

File: Services/LogService/ModsenOnlineStore.LogService.API/Controllers/LogController.cs (hunk 13 above)
Comment (Evgeny-Gromyko) [reply Services/LogService/ModsenOnlineStore.LogService.API/Controllers/LogController.cs]: removed

File: Services/LogService/ModsenOnlineStore.LogService.Infrastructure/Data/LogRepository.cs (hunk 11 above)
Comment (Evgeny-Gromyko) [reply Services/LogService/ModsenOnlineStore.LogService.Infrastructure/Data/LogRepository.cs:3]: removed

File: Services/LogService/ModsenOnlineStore.LogService.Infrastructure/Services/LogServise.cs (hunk 9 above)
Comment (Evgeny-Gromyko) [reply Services/LogService/ModsenOnlineStore.LogService.Infrastructure/Services/LogServise.cs]: removed

File: Services/LogService/ModsenOnlineStore.LogService.Infrastructure/Services/DBLogger.cs (hunk 10 above)
Comment (Evgeny-Gromyko) [reply Services/LogService/ModsenOnlineStore.LogService.Infrastructure/Services/DBLogger.cs]: removed

File: Services/Store/ModsenOnlineStore.Store.Application/Middleware/ExceptionHandlerMiddleware.cs (hunk 7 above)
Comment (Evgeny-Gromyko) [reply Services/Store/ModsenOnlineStore.Store.Application/Middleware/ExceptionHandlerMiddleware.cs]: Reminder to use logging after task approval.
Sorry, removed

**Review by elv1sabene — APPROVED** [review]

**Pull request merged**